domain/services/model.go

```diff
@@ -174,6 +174,7 @@ func (s *ModelServices) Application() *applicationservice.WatchableService {
 		secretstate.NewState(changestream.NewTxnRunnerFactory(s.modelDB), log),
 		applicationstate.NewCharmState(changestream.NewTxnRunnerFactory(s.modelDB)),
 		applicationstate.NewResourceState(changestream.NewTxnRunnerFactory(s.modelDB), s.logger.Child("resource")),
+		applicationstate.NewContainerImageMetadataState(changestream.NewTxnRunnerFactory(s.modelDB), s.logger.Child("resource")),
```

Comment:
**suggestion:** Maybe add a nested logger into resource logger ? It would be amazing and adroit, isn't it ?

Reply:
Would be cool, though sounds like with [Simon's change](https://github.com/juju/juju/pull/18422) that's not going to be possible.